docs/configuration.md

```diff
@@ -333,3 +333,25 @@ Credential: "git:https://bob@github.com/example/myrepo" (user = bob)
 
    https://bob@github.com/example/myrepo
 ```
+
+---
+
+### credential.azreposPATMode
```

Comment:
Can we get some mention of "legacy" in the variable name? While it defaults to `true` now, we'd like to default it to `false` in the not so distant future. (That might also be worth documenting here...)

Reply:
Would this be better as a set of values rather than a boolean?
I was initially thinking of just a set like `credential.azreposmode := "pat" | "azure" | ...`
but maybe instead a list of values (in case it is decided that we need to support
both old and new (and newer) ways at the same time.  (This may not make sense, but I'll ask the question.)

just in case the world changes again, or it is decided that we need to support both

Reply:
Either are good ideas. `credential.useLegacyCredentials` for example?

Or `credential.azreposAuthMode := "pat" | "azure"` etc

Reply:
I think also to make the branding gods happy, we should spell out "azureReposAuthMode". Sorry for the wordiness.

Reply:
Updated to use @jeffhostetler 's idea. I didn't rename to the full `azureRepos` because we already have several options for the Azure Repos provider that are `azRepos` and consistency is more important than trying to please brand gods on a case that was lost before it was started. 😆